include/clap/helpers/host.hh

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#include <clap/clap.h>
+
+namespace clap { namespace helpers {
+   class Host {
+   public:
+      // not copyable, not moveable
+      Host(const Host &) = delete;
+      Host(Host &&) = delete;
+      Host &operator=(const Host &) = delete;
+      Host &operator=(Host &&) = delete;
+
+      const clap_host *clapHost() const { return &_host; }
+
+   protected:
+      Host(const char *name, const char *vendor, const char *url, const char *version);
+      virtual ~Host() = default;
+
+      /////////////////////////
+      // Methods to override //
+      /////////////////////////
+
+      // clap_host
+      virtual void requestRestart() noexcept {}
+      virtual void requestProcess() noexcept {}
+      virtual void requestCallback() noexcept {}
+
+      // clap_host_audio_ports
+      virtual bool implementsAudioPorts() const noexcept { return false; }
+      bool audioPortsIsRescanFlagSupported(uint32_t flag) noexcept { return false; }
+      void audioPortsRescan(uint32_t flags) noexcept {}
+
+      // clap_host_gui
+      virtual bool implementsGui() const noexcept { return false; }
+      virtual void guiResizeHintsChanged() noexcept {}
+      virtual bool guiRequestResize(uint32_t width, uint32_t height) noexcept { return false; }
+      virtual bool guiRequestShow() noexcept { return false; }
+      virtual bool guiRequestHide() noexcept { return false; }
+      virtual void guiClosed(bool wasDestroyed) noexcept {}
+
+      // clap_host_latency
+      virtual bool implementsLatency() const noexcept { return false; }
+      static void latencyChanged() noexcept {}
+
+      // clap_host_log
+      virtual bool implementsLog() const noexcept { return false; }
+      virtual void logLog(clap_log_severity severity, const char *message) const noexcept {}
```

Comment:
default implementation here should write to `std::clog` maybe?

Reply:
Maybe `clog` isn't the right output, but something should go on the console at least by default.

Reply:
Ah actually I'm not sure, given that `implementsLog()` returns false by default.
The current code is fine :ok_hand: